frontend/src/modules/machinery/components/Machinery.js

```diff
@@ -54,6 +54,12 @@ export default class Machinery extends React.Component<Props> {
         }
     }
 
+    handleResetSearch = (event: SyntheticMouseEvent<>) => {
+        event.preventDefault();
+        this.searchInput.current.value = '';
+        this.props.searchMachinery(this.searchInput.current.value);
```

Comment:
I'd expect `this.props.searchMachinery('');` to be sufficient here, because `componentDidUpdate` should set `this.searchInput.current.value` accordingly.

And I don't think we need to prevent default?

Reply:
Yep, much cleaner to just set with an empty string.

I wasn't sure about the prevent default either. I added it to protect against an action that I might have overlooked, but it does not appear to be needed.